src/eopf_geozarr/s2_optimization/s2_multiscale.py

```diff
@@ -447,31 +461,95 @@ def add_multiscales_metadata_to_parent(
+                    log.warning(
+                        "Insufficient coordinate points for transform calculation",
+                        x_len=len(x_coords),
+                        y_len=len(y_coords),
+                        level=res_name,
+                    )
+            else:
+                log.warning(
+                    "Missing x/y coordinates for transform calculation",
+                    coords=list(dataset.coords.keys()),
+                    level=res_name,
+                )
+
         # Calculate zoom level (higher resolution = higher zoom)
         tile_width = 256
         zoom_for_width = max(0, int(np.ceil(np.log2(width / tile_width))))
         zoom_for_height = max(0, int(np.ceil(np.log2(height / tile_width))))
         zoom = max(zoom_for_width, zoom_for_height)
 
-        # Calculate scale factor relative to finest resolution
+        # Calculate relative scale and translation vs first resolution
         finest_res_meters = res_order[all_resolutions[0]]
-        scale_factor = res_meters // finest_res_meters
+        relative_scale = res_meters / finest_res_meters
+        relative_translation = (res_meters - finest_res_meters) / 2
 
-        # calculate translation relative to finest resolution
-        trans_meters = (res_meters - finest_res_meters) / 2
+        # Get chunks in the correct format
+        var_chunks = dataset.data_vars[first_var.name].chunks
+        if var_chunks is not None:
+            chunks = tuple(tuple(int(c) for c in chunk_dim) for chunk_dim in var_chunks)
+        else:
+            chunks = None
+            log.warning(
+                "Could not determine chunking information for overview level; 'chunks' will be set to None",
+                level=res_name,
+                variable=str(first_var.name),
+            )
 
-        overview_levels.append(
-            {
-                "level": res_name,  # Use string-based level name
-                "zoom": zoom,
-                "width": width,
-                "height": height,
-                "translation_relative": trans_meters,
-                "scale_absolute": res_meters,
-                "scale_relative": scale_factor,
-                "chunks": dataset.data_vars[first_var.name].chunks,
-            }
-        )
+        layout_entry = {
+            "level": res_name,  # Use string-based level name
+            "zoom": zoom,
+            "width": width,
+            "height": height,
+            "translation_relative": relative_translation,
+            "scale_absolute": res_meters,
+            "scale_relative": relative_scale,
+            "chunks": chunks,
+            "spatial_shape": [height, width],
+        }
+
+        # Only add spatial_transform if we have valid transform data
+        if transform is not None and not all(t == 0 for t in transform):
+            layout_entry["spatial_transform"] = transform
+
+        overview_levels.append(cast("OverviewLevelJSON", layout_entry))
```

Comment:
if we declare `layout_entry: OverviewLevelJSON` then we don't need this cast, and we will get type checking when we create the value